identity/src/keypair.rs

```diff
@@ -107,50 +107,14 @@ impl Keypair {
         Keypair::Ecdsa(ecdsa::Keypair::generate())
     }
 
-    #[cfg(feature = "ed25519")]
-    pub fn into_ed25519(self) -> Option<ed25519::Keypair> {
```

Comment:
> I am fine with adding `From` implementations but I want these to stay because:
> 
> a) They are easier to discover b) The compiler will struggle with type inference where there are multiple `From` or `TryFrom` implementations

Yes. `From` implementations tend to be "implicit". But the return value should be wrapped in `Result` because `Option` suggests that the value may be present, which is not the case since one instance of `Keypair` only holds one type of keys, and `Result` return type requires `try` prefix.